[matter-js] Fixed/Added ability to pass an array to Composite.remove

TaylorS15 opened the pull request:
Added ability to pass Body/Composite/Constraint/MouseConstraint array to Composite.remove. 

Composite.add has this ability and functions the same

Has been tested.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Source Code](https://github.com/liabru/matter-js/blob/master/src/body/Composite.js)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

stevensnoeijen commented:
Nice! The initial suggestion looks good, but can you also remove the type from the jsdoc param object, as this is incorrect.
Also please add a few tests related to the function that you changed, that should only be a few lines.
Thanks :)

TaylorS15 commented:
I added a test similar to the Composite.add test above it. New to testing like this and accidentally made the PR a bit messy lol. The checks still failed but not sure why, it doesn't look like it has to do with the test I added. If there's anything I need to change let me know. Thanks.

stevensnoeijen commented on types/matter-js/matter-js-tests.ts:
```diff
@@ -193,6 +193,8 @@ Composite.add(composite1, constraint1);
 // $ExpectType Composite
 Composite.add(composite3, [box1, composite2, constraint1, mouseConstraint]);
+// $ExpectType Composite
+Composite.remove(composite3, [box1, composite2, constraint1, mouseContraint]);
```
Just a small typo, otherwise this looks good :)
```suggestion
Composite.remove(composite3, [box1, composite2, constraint1, mouseConstraint]);
```

stevensnoeijen commented on types/matter-js/index.d.ts:
```diff
@@ -1385,11 +1385,20 @@ declare namespace Matter {
          * @method remove
          * @param {Composite} composite
-         * @param {any} object
+         * @param {Body | Composite | Constraint | MouseConstraint | (Body | Composite | Constraint | MouseConstraint)[]} object
```
I did expect that it would allow in jsdoc to have no type defined in the param, but I guess not? 
But this is fine too, but prefer to align the jsdoc type to the type set in the ts (Array instead of []). Bu this is optional.

```suggest
         * @param {Body | Composite | Constraint | MouseConstraint | Array<Body | Composite | Constraint | MouseConstraint>} object
 ```

stevensnoeijen reviewed (changes requested):
These suggestions should fix it, if it builds correctly we release it :)
Thanks again!

DangerBotOSS commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## matter-js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/matter-js@latest/))
was missing the following properties:
1. Matter
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 048873290a94bd119100f118ca0859a71f96d805
</p>

TaylorS15 commented:
Got it working and everything looks good to me. The test issue was just a typo as pointed out by [stevensnoeijen](https://github.com/stevensnoeijen) and just needed to fix trailing whitespace errors in index.d.ts.

stevensnoeijen commented:
Ready to merge

stevensnoeijen commented:
Will be released soon, thanks for the contribution @TaylorS15
